pkg/route/api/validation/validation.go

```diff
@@ -102,9 +108,126 @@ func validateTLS(tls *routeapi.TLSConfig) fielderrors.ValidationErrorList {
+	result := fielderrors.ValidationErrorList{}
+	if len(caCert) == 0 {
+		return result
+	}
+
+	//decode and parse the ca cert.  When a pem block is decoded the following occurs:
+	//1. if a block is parsed it is put into block
+	//2. if a block is parsed then the remaining data will be in rest
+	//3. if the data is not PEM data then block will be nil and the data will be in rest
+	block, rest := pem.Decode([]byte(caCert))
+	var parsedCACert *x509.Certificate
+	var err error
+
+	//not pem data
+	if block == nil {
+		result = append(result, fielderrors.NewFieldInvalid(field, caCert, "unable to parse certificate"))
+	} else {
+		//pem data found
+		//keep the first ca cert so we can check signatures if requested
+		parsedCACert, err = x509.ParseCertificate(block.Bytes)
+		//bad CA, don't keep parsing the chain and just return
+		if err != nil {
+			msg := fmt.Sprintf("the %s certificate was not able to be parsed: %s", field, err.Error())
+			result = append(result, fielderrors.NewFieldInvalid(field, caCert, msg))
+			return result
+		}
+
+		//check remaining rest data for more pem blocks (a cert chain concatenated in the field)
+		if len(rest) > 0 {
+			for {
+				block, rest = pem.Decode(rest)
+				//not pem data
+				if block == nil {
+					result = append(result, fielderrors.NewFieldInvalid(field, caCert, "unable to parse certificate chain"))
+					break
+				}
+				//check the cert
+				_, err := x509.ParseCertificate(block.Bytes)
+				if err != nil {
+					msg := fmt.Sprintf("the %s certificate chain was not able to be parsed: %s", field, err.Error())
+					result = append(result, fielderrors.NewFieldInvalid(field, caCert, msg))
+				}
+				//check if we're done with the pem blocks now
+				if len(rest) == 0 {
+					break
+				}
+			}
+		}
+	}
+
+	if checkSignature && parsedCACert != nil {
+		//load up the leaf cert as an x509 certificate so we can check the signatures
+		//this shouldn't die since we were able to parse it in validateTLSCertificates and tls.X509KeyPair performs this same
+		//check, it just doesn't set the .Leaf value.
+		leaf, err := x509.ParseCertificate(cert.Certificate[0])
+		if err != nil {
+			msg := fmt.Sprintf("the certificate was not able to be parsed %s", err.Error())
+			//no value set on this because at this time we have a decoded pem block, not what was originally submitted
+			result = append(result, fielderrors.NewFieldInvalid("certificate", "", msg))
+			return result
+		}
+
+		err = leaf.CheckSignatureFrom(parsedCACert)
+		if err != nil {
+			msg := fmt.Sprintf("invalid ca cert: %s", err.Error())
+			//no value set because it is a multi-field validation
+			result = append(result, fielderrors.NewFieldInvalid("certificate/caCertificate", "", msg))
+		}
+	}
+
+	return result
+}
+
+// decodeNewlines is utility to remove the json formatted newlines from a cert
+func decodeNewlines(s string) string {
```

Comment:
this is a holdover from defining a cert inside the route's json.  The newlines need escaped in the json and translate just fine when you write the file to disk.  When you try to load the json value like I'm doing though...not so much.  If there is a better way then we should definitely use it and get rid of this method.